aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1125,6 +1122,47 @@ impl AptosVM {
         )
     }
 
+    // Called when the execution of the user transaction fails, in order to discard the
+    // transaction, or clean up the failed state.
+    fn on_user_transaction_execution_failure(
```

Comment:
Should the execute script use the same cleanup function? Seems that it's only used for multi sig txn?

Reply:
Which script execution do you refer to @runtian-zhou ? So far I use it in the end of `execute_user_transaction_impl`

Reply:
Sry this looks good to me!